cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -942,6 +952,25 @@ mod tests {
 
     use super::*;
 
+    #[tokio::test]
+    async fn user_agent_set_on_all_requests() {
+        let expected_agent = format!("flox-cli/{}", &*FLOX_VERSION);
```

Comment:
> matching the version exactly in the test makes sense to me.
> wouldnt we otherwise be asserting the version schema instead? not that we intend to change that anytime soon.

It feels odd that we're duplicating the same `format` call and re-using `FLOX_VERSION` directly rather than asserting what the structure of the value should be. It's not a big deal though.